core/src/main/java/zipkin2/reporter/Sender.java

```diff
@@ -40,42 +41,13 @@
  * send messages as byte arrays.
  *
  * @since 3.0
+ * @deprecated since 3.2, use {@link BytesMessageSender} instead. This will be removed in v4.0.
  */
-public abstract class Sender extends Component {
+@Deprecated
+public abstract class Sender extends Component implements BytesMessageSender {
```

Comment:
retrofitted Sender as BytesMessageSender so that non-updated code can drop in and work. This is also why BytesMessageSender is an interface, as we can't extend two classes